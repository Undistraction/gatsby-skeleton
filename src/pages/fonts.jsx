import React from 'react';
import Page from '../components/Page/Page';
import Fonts from '../components/Fonts/Fonts';
import Metadata from '../components/Metadata/Metadata';
import loadMetadata from '../utils/loadMetadata';

const FontsPage = () => {
  const metadata = loadMetadata('fonts');
  return (
    <Page title="Fonts">
      <Metadata {...metadata} />
      <Fonts />
    </Page>
  );
};

export default FontsPage;
