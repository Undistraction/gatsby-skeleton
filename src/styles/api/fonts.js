// import isNotObject from 'ramda-adjunct';
// import { throwInvalidFontMapError } from '../errors';

// export default fontMap => {
//   if (isNotObject(fontMap)) {
//     throw new Error('');
//   }

//   return (name, { weight, style }) => {};
// };

// theme.fonts('default', normal, italic );

// theme.fonts.weightOf('default', 'bold'); // 300;
// theme.fonts.face('normal'); // Example, sans-serif;
// theme.fonts('default', 'italic')

// font-face: Example, sans-serif
// font-weight: 300,
// font-style: normal
