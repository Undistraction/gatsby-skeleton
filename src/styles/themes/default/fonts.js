export default Object.freeze({
  default: {
    stack: 'Cousine, monospace',
    weights: {
      normal: {
        weight: 400,
        styles: ['regular', 'italic'],
      },
      bold: {
        weight: 700,
        styles: ['regular', 'italic'],
      },
    },
  },
});
