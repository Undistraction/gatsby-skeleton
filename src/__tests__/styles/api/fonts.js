import fonts from '../../../styles/api/fonts';

describe('fonts', () => {
  it('errors if no font map is supplied', () => {
    expect(() => fonts()).toThrowErrorMatchingSnapshot();
  });
});
