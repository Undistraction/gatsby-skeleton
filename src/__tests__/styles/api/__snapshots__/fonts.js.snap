// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fonts errors if no font map is supplied 1`] = `"(0 , _ramdaAdjunct2.default) is not a function"`;
